homeassistant/components/climate/tado_v1.py

```diff
@@ -0,0 +1,301 @@
+
+# When we change the temperature setting, we need an overlay mode
+# wait until tado changes the mode automatic
+CONST_OVERLAY_TADO_MODE = "TADO_MODE"
+# the user has change the temperature or mode manually
+CONST_OVERLAY_MANUAL = "MANUAL"
+# the temperature will be reset after a timespan
+CONST_OVERLAY_TIMER = "TIMER"
+
+
+# will be used when changing temperature
+CONST_DEFAULT_OPERATION_MODE = CONST_OVERLAY_TADO_MODE
+# will be used when switching to CONST_MODE_OFF
+CONST_DEFAULT_OFF_MODE = CONST_OVERLAY_MANUAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the climate platform."""
+    # get the PyTado object from the hub component
+    tado = hass.data[CONST_TADO_DATA]
+
+    try:
+        zones = tado.get_zones()
+    except RuntimeError:
+        _LOGGER.error("Unable to get zone info from mytado")
+        return False
+
+    climate_devices = []
+    for zone in zones:
+        climate_devices.append(create_climate_device(tado, hass,
+                                                     zone,
+                                                     zone['name'],
+                                                     zone['id']))
+
+    if len(climate_devices) > 0:
+        add_devices(climate_devices)
+        return True
+    else:
+        return False
+
+
+def create_climate_device(tado, hass, zone, name, zone_id):
+    """Create a climate device."""
+    capabilities = tado.get_capabilities(zone_id)
+
+    min_temp = float(capabilities["temperatures"]["celsius"]["min"])
+    max_temp = float(capabilities["temperatures"]["celsius"]["max"])
+    target_temp = 21
+    ac_mode = capabilities["type"] != "HEATING"
+
+    data_id = 'zone {} {}'.format(name, zone_id)
+    device = TadoClimate(tado,
+                         name, zone_id, data_id,
+                         min_temp, max_temp,
+                         target_temp, ac_mode)
+
+    tado.add_sensor(data_id, {
+        "id": zone_id,
+        "zone": zone,
+        "name": name,
+        "climate": device
+    })
+
+    return device
+
+
+class TadoClimate(ClimateDevice):
+    """Representation of a tado climate device."""
+
+    def __init__(self, store, zone_name, zone_id, data_id,
+                 min_temp, max_temp, target_temp, ac_mode,
+                 tolerance=0.3):
+        """Initialization of TadoClimate device."""
+        self._store = store
+        self._data_id = data_id
+
+        self.zone_name = zone_name
+        self.zone_id = zone_id
+
+        self.ac_mode = ac_mode
+
+        self._active = False
+        self._device_is_active = False
+
+        self._cur_temp = None
+        self._cur_humidity = None
+        self._is_away = False
+        self._min_temp = min_temp
+        self._max_temp = max_temp
+        self._target_temp = target_temp
+        self._tolerance = tolerance
+        self._unit = TEMP_CELSIUS
+
+        self._operation_list = [CONST_OVERLAY_MANUAL,
```

Comment:
Nice work with the dict. One quick suggestion I would have is to reverse the keys and the values you have right now, then `current_operation` could just be `return OPERATION_LIST.get(self._current_operation)`.